src/diffusers/utils/hub_utils.py

```diff
@@ -129,3 +134,116 @@ def create_model_card(args, model_name):
+            if resolved_file is not _CACHED_NO_EXIST:
+                return resolved_file
+            else:
+                raise EnvironmentError(f"Could not locate {filename} inside {repo_id}.")
+
+    return hf_hub_download(repo_id, filename, *args, cache_dir=cache_dir, subfolder=subfolder, **kwargs)
+
+
+def try_to_load_from_cache(
+    repo_id: str,
+    filename: Union[str, Path, None] = None,
+    cache_dir: Union[str, Path, None] = None,
+    revision: Optional[str] = None,
+    subfolder: Optional[str] = None,
+) -> Optional[str]:
+    """
+    Explores the cache to return the latest cached folder or file for a given revision if found.
+
+    This function will not raise any exception if the folder or file in not cached.
+
+    Args:
+        cache_dir (`str` or `os.PathLike`):
+            The folder where the cached files lie.
+        repo_id (`str`):
+            The ID of the repo on huggingface.co.
+        filename (`str`, *optional*):
+            The filename to look for inside `repo_id`.
+        revision (`str`, *optional*):
+            The specific model version to use. Will default to `"main"` if it's not provided and no `commit_hash` is
+            provided either.
+
+    Returns:
+        `Optional[str]` or `_CACHED_NO_EXIST`:
+            Will return `None` if the folder or file was not cached. Otherwise:
+            - The exact path to the cached folder or file if it's found in the cache
+            - A special value `_CACHED_NO_EXIST` if the file does not exist at the given commit hash and this fact was
+              cached.
+    """
+    if revision is None:
+        revision = "main"
+
+    if subfolder is None:
+        subfolder = ""
+
+    if cache_dir is None:
+        cache_dir = DIFFUSERS_CACHE
+
+    object_id = repo_id.replace("/", "--")
+    repo_cache = os.path.join(cache_dir, f"models--{object_id}")
+    if not os.path.isdir(repo_cache):
+        # No cache for this model
+        return None
+    for folder in ["refs", "snapshots"]:
+        if not os.path.isdir(os.path.join(repo_cache, folder)):
+            return None
+
+    # Resolve refs (for instance to convert main to the associated commit sha)
+    cached_refs = os.listdir(os.path.join(repo_cache, "refs"))
+    if revision in cached_refs:
+        with open(os.path.join(repo_cache, "refs", revision)) as f:
+            revision = f.read()
+
+    cached_shas = os.listdir(os.path.join(repo_cache, "snapshots"))
+    if revision not in cached_shas:
+        # No cache for this revision and we won't try to return a random revision
+        return None
+
+    cached_folder = os.path.join(repo_cache, "snapshots", revision, subfolder)
+    cached_folder = cached_folder if os.path.isdir(cached_folder) else None
+
+    if filename is None:
+        # return cached folder if filename is None
+        return cached_folder
+
+    if os.path.isfile(os.path.join(repo_cache, ".no_exist", revision, filename)):
+        return _CACHED_NO_EXIST
```

Comment:
I don't understand this shortcut. I can't find `.no_exist` being saved to disk anywhere.

Reply:
@pcuenca This is done in `huggingface_hub` directly. If you try to download a file from the Hub, a `EntryNotFoundError` is thrown + we cache the fact that the file doesn't exist. It's used in `transformers` since sometimes the pipeline doesn't know what will be the file that exists on the repo (for example a tokenizer). So `transformers` tries to load all the possibilities. The first time it costs 1 HEAD cost per attempt but afterward it's cached.
[See documentation.](https://huggingface.co/docs/huggingface_hub/how-to-cache#noexist-advanced)

Reply:
But actually the whole `try_to_load_from_cache` logic is [already implemented in `huggingface_hub`](https://huggingface.co/docs/huggingface_hub/v0.12.1/en/package_reference/cache#huggingface_hub.try_to_load_from_cache) so it doesn't have to be duplicated in `diffusers`. If I remember correctly, it is duplicated is `transformers` because at that time it needed to support older versions of `huggingface_hub`. But I think it should also be removed from `transformers` now (I'm noting this one for me, will handle it tomorrow I think).

Reply:
Copy-paste from `transformers` - @sgugger do we need this?

Reply:
@patrickvonplaten I don't think you need this (see my comment above). It would have been useful if sometimes `diffusers` do not know which files must be loaded and try some potential filenames 1 by 1 on the Hub until it finds the correct one.

Reply:
Thanks, yes removed everything!